libs/auth/src/card.ts

```diff
@@ -90,15 +111,15 @@ export function areElectionManagerCardDetails(
 export function arePollWorkerCardDetails(
   cardDetails: CardDetails
 ): cardDetails is PollWorkerCardDetails {
-  return cardDetails.user.role === 'poll_worker';
+  return cardDetails.user?.role === 'poll_worker';
 }
 
 /**
  * A sub-type of CardStatus
  */
 export interface CardStatusReady<T = CardDetails> {
   status: 'ready';
-  cardDetails?: T;
+  cardDetails: T;
```

Comment:
We previously used an `undefined` `cardDetails` object to indicate that a card is unprogrammed or invalid. Now, we always have a `cardDetails` object when a card is inserted and registered, but set `cardDetails.user` to `undefined` when a card is unprogrammed or invalid. Many of the changes in this and subsequent commits are accounting for this change